### PR TITLE
Implement unary functions

### PR DESCRIPTION
**Pull request by czotti**
Need to merge #1 before this one.

### CODE REVIEW

**Comment by czotti** [comment]
Still need some refactoring to simply the loop in the `main.rs`.

**Comment by czotti** [comment]
@nilgoyette, @fmorency if you want to check.

File: src/elem.rs [hunk 1]
```diff
@@ -31,24 +31,47 @@ bin_operation!(Sub, sub, -);
...
-    fn abs(self) -> Self::Output;
-}
-
-impl Abs for Elem {
-    type Output = Elem;
+macro_rules! unary_operation {
+    ($trait:ident, $fct_name:ident, $op:path) => {
+        pub trait $trait {
+            type Output;
+            #[must_use]
```
Comment (nilgoyette) [inline src/elem.rs]: Try removing this and see what happens.

**Review by nilgoyette — APPROVED** [review]
I tried thinking about something to make the code more exciting but I can't find anything cool.

**Pull request merged**